boltdesignsystem/bolt
# Feature/editor save events

## Jira

https://app.asana.com/0/1126340469288208/1145996329931297/f

## Summary

Editor close now awaits response from drupal.  

## Details

Previously, editor save events utilized a Custom Event, and the editor closes with no knowledge or feedback on it's success.   This approach awaits a function attached to the window object, allowing the editor to reliably save knowing that the drupal update was successful, or to display an error and remain open if it fails.  

There is an example function provided which includes asyc simulation.  It should be overwritten by Drupal.

## How to test

Open the micro journeys page in pattern lab, edit a page and then save.  To see a failure, update  `const shouldThisFailToDemo = false;` on line 68 in `/packages/editor/src index.js` and repeat the process.
